worker/src/RTC/Consumer.cpp

```diff
@@ -267,6 +272,38 @@ namespace RTC
 			return;
 		}
 
+		// Check whether sequence number and timestamp sync is required.
+		if (this->syncRequired)
+		{
+			this->seqNum += 1;
+
+			uint64_t now = DepLibUV::GetTime();
+
+			if (now > this->rtpTimestamp)
+				this->rtpTimestamp = now;
+
+			this->syncRequired = false;
+
+			RequestFullFrame();
+		}
+		else
+		{
+			this->seqNum += packet->GetSequenceNumber() - this->lastRecvSeqNum;
+			this->rtpTimestamp += packet->GetTimestamp() - this->lastRecvRtpTimestamp;
+		}
+
+		// Save the received sequence number.
+		this->lastRecvSeqNum = packet->GetSequenceNumber();
+
+		// Save the received timestamp.
+		this->lastRecvRtpTimestamp = packet->GetTimestamp();
+
+		// Update packet sequence number.
+		packet->SetSequenceNumber(this->seqNum);
+
+		// Update packet timestamp.
+		packet->SetTimestamp(this->rtpTimestamp);
+
 		// Process the packet.
 		if (!this->rtpStream->ReceivePacket(packet))
 			return;
```

Comment:
If, for any reason, `rtpStream->ReceivePacket()` returns false, the seq and timestamp of the packet won't be reverted so other Consumers will fail. seq and timestamp should always be reverted, no matter what happens in each Consumer.

Reply:
True